### PR TITLE
Fix replays crashing and not finding their associated beatmap

### PR DESCRIPTION
**Pull request by tgi74**
- Pass RuleSet argument to ScoreStore (prevent Null Exception)
- Compute and store MD5 for legacy replays compatibility (compare old hash with MD5)

Fixes Exception on replay importing, and fix the "You don't have the beatmap" message appearing even if the beatmap IS present

### CODE REVIEW

**Comment by Nabile-Rahmani** [comment]
That's probably going to clash with the newer format.
Perhaps you could make a `virtual void ComputeHash(Stream)` in `BeatmapInfo` to call in import, and make a `LegacyBeatmapInfo` actually compute the MD5 hash, instead of splitting the `Hash` property.

**Comment by peppy** [comment]
Please name your branch something that isn't `master` when making contributions.

File: osu.Game/Beatmaps/BeatmapInfo.cs [hunk 1]
```diff
@@ -52,6 +52,8 @@ public class BeatmapInfo : IEquatable<BeatmapInfo>, IJsonSerializable
         [JsonProperty("file_md5")]
```
Comment (peppy) [inline osu.Game/Beatmaps/BeatmapInfo.cs]: this property should be moved

**Comment by peppy** [comment]
Also I believe this may require a database migration. did you check to see whether it automatically added the column or not?

**Review by peppy — APPROVED** [review]

**Pull request merged**